Forum post (thread) message_count updates (#5206)
hammerandchisel/discord-api-docs: docs/resources/Channel.md

| last_pin_timestamp?            | ?ISO8601 timestamp                                                         | when the last pinned message was pinned. This may be `null` in events such as `GUILD_CREATE` when a message is not pinned.                                                                      |
| rtc_region?                    | ?string                                                                    | [voice region](#DOCS_RESOURCES_VOICE/voice-region-object) id for the voice channel, automatic when set to null                                                                                  |
| video_quality_mode?            | integer                                                                    | the camera [video quality mode](#DOCS_RESOURCES_CHANNEL/channel-object-video-quality-modes) of the voice channel, 1 when not present                                                            |

> | message_count?                 | integer                                                                    | count of messages (not including starter message and deleted messsages) in a thread, stops counting at 100_000 (if the thread is created before July 1, 2022, it stops at 50)        |

Reviewer: ```suggestion
| message_count?                 | integer                                                                    | count of messages (not including starter message and deleted messages) in a thread, stops counting at 100000 (if the thread is created before July 1, 2022, it stops at 50)        |
```

Reviewer 2: it's easier to read with the underscore

Reviewer 3: ```suggestion
| message_count?                 | integer                                                                    | total number of messages in the thread excluding starter message, stops counting at 100_000 (or if the thread was created before 1 July 2022, 50)        |
```

> | message_count?                 | integer                                                                    | count of messages (not including starter message and deleted messsages) in a thread, stops counting at 100_000 (if the thread is created before July 1, 2022, it stops at 50)        |

Reviewer 4: I don't think we should document the upper limit, it's something that can change in the future, you've even expressed desires to change it, and there's always push back when changing documented things. It's probably fine t say it'll always be a 32 bit integer though

| default_auto_archive_duration? | integer                                                                    | default duration that the clients (not the API) will use for newly created threads, in minutes, to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080 |
| permissions?                   | string                                                                     | computed permissions for the invoking user in the channel, including overwrites, only included when part of the `resolved` data received on a slash command interaction                         |
| flags?                         | integer                                                                    | [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) combined as a [bitfield](https://en.wikipedia.org/wiki/Bit_field)                                                         |

> | total_message_sent?            | integer                                                                    | count of total messages ever sent in a thread, it's similar to `message_count` on message creation, but decrement on message deletion. |

Reviewer: ```suggestion
| total_message_sent?            | integer                                                                    | count of total messages ever sent in a thread, it's similar to `message_count` on message creation, but decrement on message deletion |
```

> | total_message_sent?            | integer                                                                    | count of total messages ever sent in a thread, it's similar to `message_count` on message creation, but decrement on message deletion. |

Reviewer 4: Isn't this wrong, it's never decremented, right?

Author: missed the "will not" 🤦

> | total_message_sent?            | integer                                                                    | count of total messages ever sent in a thread, it's similar to `message_count` on message creation, but decrement on message deletion. |

Reviewer 3: ```suggestion
| total_message_sent?            | integer                                                                    | count of all messages ever sent in the thread, does not decrement on message deletion |
```

Also, shouldn't this be named `total_messages_sent` as a plural?

> | message_count?                 | integer                                                                    | amount of messages (not including the initial message or deleted messages) in a thread (if the thread was created before July 1, 2022, it stops counting at 50)        |
> | member_count?                  | integer                                                                    | an approximate count of users in a thread, stops counting at 50                   |
> | thread_metadata?               | a [thread metadata](#DOCS_RESOURCES_CHANNEL/thread-metadata-object) object | thread-specific fields not needed by other channels                                                                                                                                             |
> | member?                        | a [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) object     | thread member object for the current user, if they have joined the thread, only included on certain API endpoints                                                                               |
> | default_auto_archive_duration? | integer                                                                    | default duration that the clients (not the API) will use for newly created threads, in minutes, to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080 |
> | permissions?                   | string                                                                     | computed permissions for the invoking user in the channel, including overwrites, only included when part of the `resolved` data received on a slash command interaction                         |
> | flags?                         | integer                                                                    | [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) combined as a [bitfield](https://en.wikipedia.org/wiki/Bit_field)                                                         |
> | total_message_sent?            | integer                                                                    | amount of messages ever sent in a thread, it's similar to `message_count` on message creation, but will not decrement the amount when a message is deleted            |

Reviewer 2: ```suggestion
| message_count?                 | integer                                                                    | amount of messages (not including the initial message or deleted messages) in a thread (if the thread was created before July 1, 2022, it stops counting at 50)                                 |
| member_count?                  | integer                                                                    | an approximate count of users in a thread, stops counting at 50                                                                                                                                 |
| thread_metadata?               | a [thread metadata](#DOCS_RESOURCES_CHANNEL/thread-metadata-object) object | thread-specific fields not needed by other channels                                                                                                                                             |
| member?                        | a [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) object     | thread member object for the current user, if they have joined the thread, only included on certain API endpoints                                                                               |
| default_auto_archive_duration? | integer                                                                    | default duration that the clients (not the API) will use for newly created threads, in minutes, to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080 |
| permissions?                   | string                                                                     | computed permissions for the invoking user in the channel, including overwrites, only included when part of the `resolved` data received on a slash command interaction                         |
| flags?                         | integer                                                                    | [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) combined as a [bitfield](https://en.wikipedia.org/wiki/Bit_field)                                                         |
| total_message_sent?            | integer                                                                    | amount of messages ever sent in a thread, it's similar to `message_count` on message creation, but will not decrement the amount when a message is deleted                                      |
```
table formatting

| components?               | Array of [message components](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/component-object)                                              | sent if the message contains components like buttons, action rows, or other interactive components                                               |
| sticker_items?            | array of [message sticker item objects](#DOCS_RESOURCES_STICKER/sticker-item-object)                                               | sent if the message contains stickers                                                                                                            |
| stickers?                 | array of [sticker](#DOCS_RESOURCES_STICKER/sticker-object) objects                                                                 | **Deprecated** the stickers sent with the message                                                                                                |

> | position?                 | integer                                                                                                                            | position of the message in a forum post when it was sent, it will not change upon any message deletion in the parent post                    |

Reviewer 4: I think this is for all threads, not just forum posts, right?
I'd suggest: "A generally increasing integer (there may be gaps or duplicates) that represents the approximate position of the message in a thread."
